src/Featurizers/Components/DocumentStatisticsEstimator.h

```diff
@@ -208,27 +256,32 @@ using DocumentStatisticsEstimator           = Components::TrainingOnlyEstimatorI
 
 // ----------------------------------------------------------------------
 // |
-// |  DocumentStatisticsAnnotationData
+// |  FrequencyAndIndexStruct
 // |
 // ----------------------------------------------------------------------
-inline DocumentStatisticsAnnotationData::DocumentStatisticsAnnotationData(FrequencyMap termFreq, IndexMap termIndex, std::uint32_t totalNumDocuments) :
-    TermFrequency(
-        std::move(
-            [&termFreq](void) -> FrequencyMap & {
-                if(termFreq.empty())
-                    throw std::invalid_argument("termFreq");
+FrequencyAndIndexStruct::FrequencyAndIndexStruct(std::uint32_t termFrequency, std::uint32_t index) :
+    //validate not necessary in this assignment
+    TermFrequency(std::move(termFrequency)),
+    Index(std::move(index)) {
+}
 
-                return termFreq;
-            }()
-        )
-    ),
-    TermIndex(
+bool FrequencyAndIndexStruct::operator==(FrequencyAndIndexStruct const &other) const {
+    return (TermFrequency == other.TermFrequency) && (Index == other.Index);
+}
+
+// ----------------------------------------------------------------------
+// |
+// |  DocumentStatisticsAnnotationData
+// |
+// ----------------------------------------------------------------------
+inline DocumentStatisticsAnnotationData::DocumentStatisticsAnnotationData(FrequencyAndIndexMap termFrequencyAndIndex, std::uint32_t totalNumDocuments) :
```

Comment:
>FrequencyAndIndexMap termFrequencyAndIndex, std::uint32_t totalNumDocuments [](start = 74, length = 75)

Is there a relationship between the values in `termFrequencyAndIndex` and `totalNumDocuments`? For example, should there ever be a `termFrequency` value that is > `totalNumDocuments`?

Reply:
no relationship